selftests/unit/test_nrunner.py

```diff
@@ -220,6 +224,63 @@ def test_runner_python_unittest(self):
         self.assertTrue(result['output'].startswith(output1))
         self.assertTrue(result['output'].endswith(output2))
 
+    def test_runner_tap_fail(self):
```

Comment:
It may sound a bit of an overkill, but for increased portability, it's a good idea to check for the binaries (even this basic one) on the system before assuming you can use it.  In this case, a `@skipUnless(os.path.exists("/bin/sh"))` should do the trick.

One other option, is to write Python scripts instead, and rely on `sys.executable` which is guaranteed to be present.